### PR TITLE
[SPARK-44831][PYTHON][DOCS] Refine DocString of `DataFrame.{union, unionAll, unionByName}`

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Refine DocString of `Union*`:

1. fix minor grammar mistakes
2. add more examples

### Why are the changes needed?
to improve the docs

### Does this PR introduce _any_ user-facing change?
yes

### How was this patch tested?
CI

### CODE REVIEW

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -3752,46 +3752,96 @@ def union(self, other: "DataFrame") -> "DataFrame":
...
+        This method performs a SQL-style set union of the rows from both `DataFrame` objects,
+        with no automatic deduplication of elements.
 
-        Also as standard in SQL, this function resolves columns by position (not by name).
+        Use the `distinct()` method to perform deduplication of rows.
+
+        The method resolves columns by position (not by name), following the standard behavior
+        in SQL.
+
+        Alias: The `union` method was previously named `unionAll` in versions before 2.0.0.
```
Comment (allisonwang-db) [inline python/pyspark/sql/dataframe.py:3776]: @zhengruifeng do we need to mention this?

**Review by allisonwang-db — APPROVED** [review]
Looks good! Just want to make sure we have doctest for the examples right?

File: python/pyspark/sql/dataframe.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/sql/dataframe.py:3776]: emm, seems it's not needed

**Review by srowen — APPROVED** [review]

**Comment by zhengruifeng** [comment]
thanks @srowen @allisonwang-db for review, merged to master and branch-3.5

**Pull request closed without merging**